drivers/saul/gpio_saul.c

```diff
@@ -23,12 +23,16 @@
 #include "saul.h"
 #include "phydat.h"
 #include "periph/gpio.h"
+#include "saul/periph.h"
 
 
 static int read(const void *dev, phydat_t *res)
 {
-    gpio_t pin = *((const gpio_t *)dev);
-    res->val[0] = (gpio_read(pin)) ? 1 : 0;
+    const saul_gpio_params_t *p = (const saul_gpio_params_t *)dev;
+    int inverted = (p->flags | SAUL_GPIO_INVERTED);
```

Comment:
should be `int inverted = (p->flags & SAUL_GPIO_INVERTED);`, or not?

Reply:
I do this mistake all the time.